### PR TITLE
Feat/link create route stages

### PR DESCRIPTION
**Pull request by Parle648**
## Describe your changes

- Create logic for create route

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://codecraftersintership.atlassian.net/browse/SCRUM-48)

### **General**

- [x] types for input and output params
- [x] no `any`, should be also added to eslint rules
- [x] try/catch for any async function
- [x] no **magic** [numbers](/7d77574ee4bc4e339a68066762e887cb)
- [x] compare only with constants not with strings (instead of user === ‘admin’, better user === roles.admin)
- [x] no ternary operator inside ternary operator (bad example: question ? first: second ? third: fourth)
- [x] avoid similar types with duplicating by generics
- [x] no **console.log** in pr
- [x] .env file should be in .gitignore
- [x] delete commented code if it's not part of documentation
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file
- [x] **constants** and **function** names should be lowercase.
- [x] no dates format in the code (like "dd/MM/yyyy”), move it in global constant variable
- [x] import rules. imports should come in a specific order: node modules, absolute path, relative path
- [x] strings should be in the constant variable. Example: instead of ‘15 3 \* \* _’, const EACH_DAY_15h_15min = ‘15 3 _ \* \*’

### Backend

- [x] swagger for each endpoint. add in the documentation, different types of responses.
      on a successful response (2xx), on unsuccessful response (4xx, 5xx)
- [x] less requests to db, all data should be taken with one query
- [x] **public** in method use only is use function externally
- [x] use ConfigService instead of **process.env**
- [x] use @`@index` decorator for frequently requested data
- [x] use transactions if there is a call chain that mutates data in different tables
- [x] add a set of rules for [nestjs](https://github.com/darraghoriordan/eslint-plugin-nestjs-typed) to the .eslint config file. It helps to prevent bugs and confusing moments
- [x] use [REST API Naming Convention](https://medium.com/@nadinCodeHat/rest-api-naming-conventions-and-best-practices-1c4e781eb6a5) for endpoints

  P.S: more information about RESTful API design by [Microsof](https://learn.microsoft.com/en-us/azure/architecture/best-practices/api-design)t

- [x] use ‘**UUIDs**` for primary key, ** to prevent Enumeration Attacks**

### CODE REVIEW

File: src/modules/orders/orders.service.ts [hunk 1]
```diff
@@ -133,9 +146,20 @@ export class OrdersService {
         acc[collectionDateKey] = (acc[collectionDateKey] || 0) + 1;
         return acc;
       }, {});
+    } catch (error: unknown) {
+      throw new InternalServerErrorException(error);
+    }
+  }
+
+  async findByIds(arrayOfId: string): Promise<Order[]> {
```
Comment (lyudad) [inline src/modules/orders/orders.service.ts]: why array is string?

File: src/modules/orders/orders.controller.ts [hunk 2]
```diff
@@ -60,4 +61,21 @@ export class OrdersController {
   async findOrdersDates(@Query() { date, companyId }: { date: Date; companyId: number }): Promise<Record<string, number>> {
     return this.ordersService.getDates(date, companyId);
   }
+
+  @Get('by-id')
```
Comment (lyudad) [inline src/modules/orders/orders.controller.ts]: by-ids

File: src/modules/route/dto/create-route.dto.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { Company } from 'common/database/entities/company.entity';
+import { User } from 'common/database/entities/user.entity';
+import { RouteStatuses } from 'common/enums/enums';
+
+export class CreateRouteDto {
+  submission_date: Date;
+  arrival_date: Date;
+  distance: number;
+  status: RouteStatuses;
+  user_id: User;
```
Comment (lyudad) [inline src/modules/route/dto/create-route.dto.ts:10]: why user_id is User and not a number, it look over complicated. its definitely not comfortable for frontend

File: src/modules/orders/orders.controller.ts (hunk 2 above)
Comment (Parle648) [reply src/modules/orders/orders.controller.ts]: done

File: src/modules/route/dto/create-route.dto.ts (hunk 3 above)
Comment (Parle648) [reply src/modules/route/dto/create-route.dto.ts:10]: replace by number

**Review by AlexeyBezrebriy — APPROVED** [review]

File: src/modules/orders/orders.service.ts (hunk 1 above)
Comment (Parle648) [reply src/modules/orders/orders.service.ts]: fixed

File: src/modules/orders/orders.service.ts [hunk 4]
```diff
@@ -21,14 +21,30 @@ export class OrdersService {
     companyId,
     search,
     isNew,
+    startDate,
   }: OrderServiceParams): Promise<{ orders: Order[]; page: number; pagesCount: number }> {
+    let dateFromServer;
+    let startOfDay;
+    let endOfDay;
+
+    if (startDate !== undefined) {
```
Comment (lyudad) [inline src/modules/orders/orders.service.ts]: if (!startDate)

**Pull request merged**